presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -575,18 +605,27 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
                             entry.getValue(),
                             builder.build(),
                             Optional.empty(),
-                            ImmutableList.of()));
+                            aggregations.values().stream()
+                                    .map(Aggregation::getOrderBy)
+                                    .flatMap(Collection::stream)
+                                    .collect(toImmutableList())));
         }
 
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
+                    node,
+                    "ORDER BY in aggregate function with empty grouping set");
```

Comment:
I forgot the details. But when I tested it, the query failed with an error message similar to that. I can double check. Basically I added this check so if a query would have failed without this, now it would failed with a clearer message.